src/utils/nps.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// Loosely adapted from https://github.com/microsoft/vscode-azure-account/blob/2f497562cab5f3db09f983ab5101040f27dceb70/src/nps.ts
+
+import { env, Memento, MessageItem, Uri, window } from "vscode";
+import { ext } from "vscode-azureappservice/out/src/extensionVariables";
+
+const PROBABILITY = 0.15;
+const MIN_SESSION_COUNT = 10;
+
+const SURVEY_NAME = 'nps1';
+const SURVEY_URL = 'https://www.surveymonkey.com/r/vscodedockernpsinproduct';

```

Comment:
Should be `aka.ms` link?

Reply:
I'm not sure. The Azure Account NPS survey used the direct link, and attached platform, extension version, and machine ID to the query string; I'm not sure if the aka.ms one will keep that query string for the redirected URL?

Reply:
(Assuming such a link would work with the query parameters added later.)